src/test/ui/parser/float-literals.rs

```diff
@@ -0,0 +1,12 @@
+// run-pass
+// ignore-tidy-linelength
+// Regression test for #31109 and #31407.
+// This test should compile and produce expected output.
+
+pub fn main() {
+    let n: f64 = 0.3333333333333333333333333333333333333333333333333333333333333333333333333333333333333333333333333333333333333333333333333333333333333333333333333333333333333333333333333333333333333333333333333333333333333333333333333333333333333333333333333333333333333333333333333333333333333333333333333333333333333333333333333333333333333333333333333333333333333333333333333333333333;
+    assert_eq!(n.to_bits(), 0x3fd5555555555555);
+
+    let n: f64 = 1234567890123456789012345678901234567890e-340;
+    assert_eq!(n.to_bits(), 0x1752a64e34ba0d3);
```

Comment:
Could this be a unit test instead? That would avoid having to build the compiler just to test changes to the standard library.

Reply:
It could be, I have unittests for similar behavior, however, as this replaces a few compiler tests where there was an expected failure, I'd like to make sure regressions don't occur for any reason. Specifically [issue-31109.rs](https://github.com/Alexhuszagh/rust/blob/868c702d0c9a471a28fb55f0148eb1e3e8b1dcc5/src/test/ui/issues/issue-31109.rs) and [issue-68396-let-float-bug.rs](https://github.com/Alexhuszagh/rust/blob/868c702d0c9a471a28fb55f0148eb1e3e8b1dcc5/src/test/ui/pattern/issue-68396-let-float-bug.rs).

Reply:
I don't think making it a compiler test will help. The tests were making sure that the compiler gracefully failed on errors, not that the parsing was correct. Since there aren't any errors, there's nothing to test.

Reply:
Removed the compiler test, since it intrinsically depends on `dec2flt` anyway.